### PR TITLE
db: addition of versioning support

### PR DESCRIPTION
**Pull request by SamiHiltunen**
* Adds SQLAlchemy-Continuum support to Invenio-DB for
  versioning support.

Signed-off-by: Sami Hiltunen <sami.mikael.hiltunen@cern.ch>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -69,8 +69,9 @@
 install_requires = [
     'Flask-CLI>=0.2.1',
     'Flask-SQLAlchemy>=2.0',
-    'SQLAlchemy-Utils>=0.31.0',
     'SQLAlchemy>=1.0',
+    'SQLAlchemy-Utils>=0.31.0',
+    'SQLAlchemy-Continuum>=1.2.1',
```
Comment (jirikuncar) [inline setup.py]: this should be an optional extra

File: invenio_db/core.py [hunk 2]
```diff
@@ -59,6 +61,9 @@ def init_db(self, app, entrypoint_name='invenio_db.models', **kwargs):
             'sqlite:///' + os.path.join(app.instance_path, app.name + '.db')
         )
         app.config.setdefault('SQLALCHEMY_ECHO', app.debug)
+        app.config.setdefault('DB_VERSIONING_USER_MODEL', None)
+        with app.app_context():
+            make_versioned(user_cls=app.config['DB_VERSIONING_USER_MODEL'])
```
Comment (jirikuncar) [inline invenio_db/core.py]: you might need to import the user model and do it after `db.init_app(app)`

**Comment by jirikuncar** [comment]
see also https://github.com/inveniosoftware/invenio-db/issues/24

File: invenio_db/core.py [hunk 3]
```diff
@@ -59,6 +61,9 @@ def init_db(self, app, entrypoint_name='invenio_db.models', **kwargs):
             'sqlite:///' + os.path.join(app.instance_path, app.name + '.db')
         )
         app.config.setdefault('SQLALCHEMY_ECHO', app.debug)
+        app.config.setdefault('DB_VERSIONING_USER_MODEL', None)
```
Comment (jirikuncar) [inline invenio_db/core.py]: We might have troubles with the initialization order. The only way how to resolve it, which I can think of right now, is to use entry point.

```python
'invenio_db.versioned': [
    'user_cls = invenio_accounts.models:User',
],
```

You can try to load this entry point if there is nothing in config 'DB_VERSIONING_USER_MODEL'

Comment (jirikuncar) [inline invenio_db/core.py]: @lnielsen do you have any better idea then https://github.com/inveniosoftware/invenio-db/pull/21/files#r47119393 ?

Comment (jirikuncar) [inline invenio_db/core.py]: After musings with @lnielsen the prefered way is still this one if the `DB_VERSIONING_USER_MODEL` value is not set:
```python
try:
    pkg_resources.get_distribution('invenio_accounts')
except pkg_resources.DistributionNotFound:
    user_cls = None
else:
    from invenio_accounts.models import User as user_cls
```

**Comment by SamiHiltunen** [comment]
Closing as most of the changes have been merged in with #25.

**Pull request closed without merging**